### PR TITLE
feat(protocol/protocol-2.0): Update contract descriptions to Bedrock

### PR DESCRIPTION
**Pull request by qbzzt**
Closing: DEVRL-981

### CODE REVIEW

File: src/docs/protocol/protocol-2.0.md [hunk 1]
```diff
@@ -3,164 +3,86 @@ title: Contract Overview
...
+This is the contract that replaces the old State Commitment Chain.
 
-Optimism is meant to look, feel and behave like Ethereum but cheaper and faster.
-For developers building on our Optimism, we aim to make the transition as seamless as possible.
-With very few exceptions,
-existing Solidity smart contracts can run on L2 exactly how they run on L1.
-Similarly, off-chain code (ie. UIs and wallets), should be able to interact with L2 contract with little more than an updated RPC endpoint.
+### OptimismPortal
 
+[The `OptimismPortal` contract](https://github.com/ethereum-optimism/optimism/blob/65ec61dde94ffa93342728d324fecf474d228e1f/packages/contracts-bedrock/contracts/L1/OptimismPortal.sol) provides [the low-level API for communications between layers](#deposits-from-the-underlying-l1-ethereum-goerli-etc-to-l2-op-mainnet-op-goerli-etc). Unless you are trying to send L2 transactions via L1 to bypass the sequencer, we strongly recommend sending messages between L1 and L2 via the L1CrossDomainMessenger and L2CrossDomainMessenger.
```
Comment (sbvegan) [inline src/docs/protocol/protocol-2.0.md:17]: [the low-level API for communications between layers](#deposits-from-the-underlying-l1-ethereum-goerli-etc-to-l2-op-mainnet-op-goerli-etc) doesn't take the user anywhere

File: src/docs/protocol/protocol-2.0.md [hunk 2]
```diff
@@ -3,164 +3,86 @@ title: Contract Overview
 lang: en-US
 ---
 
-## Introduction
+## L1 contracts
 
-<!-- - Welcome!  Give context -- "how to read these docs" -->
+### L2OutputOracle
```
Comment (sbvegan) [inline src/docs/protocol/protocol-2.0.md:8]: Do you think its worth mention these other L1 contracts?
- https://github.com/ethereum-optimism/optimism/blob/65ec61dde94ffa93342728d324fecf474d228e1f/packages/contracts-bedrock/contracts/L1/L1ERC721Bridge.sol
- https://github.com/ethereum-optimism/optimism/blob/65ec61dde94ffa93342728d324fecf474d228e1f/packages/contracts-bedrock/contracts/L1/SystemConfig.sol
- https://github.com/ethereum-optimism/optimism/blob/65ec61dde94ffa93342728d324fecf474d228e1f/packages/contracts-bedrock/contracts/L1/ResourceMetering.sol

File: src/docs/protocol/protocol-2.0.md [hunk 3]
```diff
@@ -3,164 +3,86 @@ title: Contract Overview
...
-- **[Verification:](#verification)** Contracts on layer-1 which implement the process for challenging a transaction result.
-- **[Bridge:](#bridge-contracts)** Contracts which facilitate message passing between layer-1 and layer-2.
-- **[Predeploys:](#predeployed-contracts)** A set of essential contracts which are deployed and available in the genesis state of the system. These contracts are similar to Ethereum's precompiles, however they are written in Solidity, and can be found at addresses prefixed with 0x42.
+### L1StandardBridge
 
+[The `L1StandardBridge` contract](https://github.com/ethereum-optimism/optimism/blob/65ec61dde94ffa93342728d324fecf474d228e1f/packages/contracts-bedrock/contracts/L1/L1StandardBridge.sol) uses `L1CrossDomainMessenger` to transfer ETH and ERC-20 tokens between the underlying L1 (Ethereum, Goerli, etc.) and L2 (OP Mainnet, OP Goerli, etc.).
 
 
-## Chain Contracts
+## L2 contracts (predeploys)
```
Comment (sbvegan) [inline src/docs/protocol/protocol-2.0.md:28]: Do you think its worth mentioning any of these L2 contacts:
- https://github.com/ethereum-optimism/optimism/blob/65ec61dde94ffa93342728d324fecf474d228e1f/packages/contracts-bedrock/contracts/L2/BaseFeeVault.sol
- https://github.com/ethereum-optimism/optimism/blob/65ec61dde94ffa93342728d324fecf474d228e1f/packages/contracts-bedrock/contracts/L2/CrossDomainOwnable.sol
- https://github.com/ethereum-optimism/optimism/blob/65ec61dde94ffa93342728d324fecf474d228e1f/packages/contracts-bedrock/contracts/L2/CrossDomainOwnable2.sol
- https://github.com/ethereum-optimism/optimism/blob/65ec61dde94ffa93342728d324fecf474d228e1f/packages/contracts-bedrock/contracts/L2/CrossDomainOwnable3.sol
- https://github.com/ethereum-optimism/optimism/blob/65ec61dde94ffa93342728d324fecf474d228e1f/packages/contracts-bedrock/contracts/L2/GasPriceOracle.sol
- https://github.com/ethereum-optimism/optimism/blob/65ec61dde94ffa93342728d324fecf474d228e1f/packages/contracts-bedrock/contracts/L2/L1FeeVault.sol
- https://github.com/ethereum-optimism/optimism/blob/65ec61dde94ffa93342728d324fecf474d228e1f/packages/contracts-bedrock/contracts/L2/L2ERC721Bridge.sol

File: src/docs/protocol/protocol-2.0.md (hunk 2 above)
Comment (qbzzt) [reply src/docs/protocol/protocol-2.0.md:8]: You're right that we should add those contract descriptions. However, that should be a separate update so we can merge this already.

**Review by sbvegan — APPROVED** [review]

File: src/docs/protocol/protocol-2.0.md (hunk 2 above)
Comment (sbvegan) [reply src/docs/protocol/protocol-2.0.md:8]: Sounds good

**Review by OPMattie — APPROVED** [review]

**Pull request merged**